### ISSUE TITLE
[BUG] [Formatter] Meaningful whitespace added around `filter` tag

### ISSUE THREAD

**Issue by dyve** [Riverside-Healthcare/djLint#165]
- [ ] I'm on the [latest version](https://pypi.org/project/djlint/) of djLint
- [ ] I've searched the [issues](https://github.com/Riverside-Healthcare/djLint/issues)
- [ ] I've read the [docs](https://djlint.com)

## System Info
 - OS: macOS 12.1
 - Python Version: 3.8.12
 - djLint Version: 0.7.0

## Issue
Meaningful whitespace added around `filter` tag

## How To Reproduce

Use djlint to reformat this file:

```
{% filter capfirst %}{% blocktrans %}dear {{ name }}{% endblocktrans %}{% endfilter %},

How are you?
```

Which then turns into:

```
{% filter capfirst %}
    {% blocktrans %}dear {{ name }}{% endblocktrans %}
{% endfilter %}
,
How are you?
```

The added whitespace is meaningful, it shows ub between `{{name }}` and the comma.

## Configuration

[tool.djlint]
# Reference: https://djlint.com/djlint/rules.html
profile = "django"
ignore = "H006,H008,H013,H014,H017,H019,H021,H023,T001,T002,T003"
blank_line_after_tag = "load,extends,endblock,has_perm"
max_line_length = 120

**Comment by dyve** [comment]
Might be pushing the limits here. Whitespace in HTML is nasty (see other issues). This might also be solved by

```
{# djlint: off #}
{% filter capfirst %}{% blocktrans %}dear {{ name }}{% endblocktrans %}{% endfilter %},
{# djlint: on #}
```

**Comment by christopherpickering** [comment]
Yeah this is tricky, I'm surprised the output html has the extra space. What is the surrounding html tag and style? Is it a white-space:pre or something?

**Comment by dyve** [comment]
I reduced it to the smallest example, if you make a `temp.html` with the content from the issue it is enough to trigger the effect. The underlying problem is that almost all whitespace in HTML (templates) is meaningful. Vertical whitespace for formatting is also horizontal whitespace after rendering the HTML. Especially template tag systems that do not have whitespace absorbing tags like `{%- -%}` in Jinja2 have to be very careful.

**Comment by christopherpickering** [comment]
I wonder if the correct solution would be that when djlint is adding line breaks after closing tags (</div> or {% endsomething %}) that we do not add a break if it is followed directly by text or a horizontal space + text.

**Comment by dyve** [comment]
A horizontal space and a break are both whitespace in HTML. The only time you cannot replace a horizontal space with a break is if you are in `pre` mode.

I don't you you can catch every HTML template code, so it's important that `djlint:on` and 'djlint:off` work properly.

Maybe differ behavior between block and inline tags, even though CSS can overwrite that.

Maybe add `djlint:break:on`and `djlint:break:off`, to allow or disallow djlint to insert breaks.

Other than that, define a sensible default and let the user sort it out if they use crazy markup (which is sometimes necessary, but should then be done on purpose).

**Comment by zyzzyxdonta** [comment]
This also seems to be happening with double quotes/before ifs. See lines 11 and 13 in this example.
[import_from.html.txt](https://github.com/Riverside-Healthcare/djLint/files/11234098/import_from.html.txt)
